apps/site/assets/ts/trip-planner/__tests__/TripPlannerPageTest.tsx

```diff
@@ -0,0 +1,201 @@
+import React from "react";
+import renderer, { act } from "react-test-renderer";
+import {
+  MapData,
+  MapMarker as Marker
+} from "../../leaflet/components/__mapdata";
+import { createReactRoot } from "../../app/helpers/testUtils";
+import TripPlannerPage, {
+  Action,
+  RemoveEvent,
+  State,
+  UpdateEvent,
+  onRemove,
+  onUpdate,
+  reducer
+} from "../components/TripPlannerPage";
+const mapData: MapData = {
+  default_center: { latitude: 0, longitude: 0 },
+  height: 500,
+  markers: [],
+  polylines: [],
+  tile_server_url: "",
+  width: 630,
+  zoom: 14
+};
+
+describe("TripPlannerPage", () => {
+  it("renders", () => {
+    createReactRoot();
+
+    const tree = renderer
+      .create(<TripPlannerPage mapData={mapData} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("registers event listeners", () => {
+    const cbHash: { [event: string]: Function } = {};
+    document.addEventListener = (name: string, cb: EventListener) => {
+      cbHash[name] = cb;
+    };
+    renderer
+      .create(<TripPlannerPage mapData={mapData} />)
+      // update triggers useEffect to run
+      .update(<TripPlannerPage mapData={mapData} />);
+    expect(Object.keys(cbHash)).toEqual([
```

Comment:
Could this `expect` be after line 65? I'm a little confused by the code below not having an assertion.

Reply:
using the `act` functions got the code coverage to recognize that these were being called (and therefore increase the code coverage), but I wasn't able to figure out how to verify that things changed as expected. I feel pretty confident because of the unit tests. Do you know if there's a way to check the component's state from outside?

Reply:
We discussed offline that we can't test the preferred way because the trip plan events aren't being triggered by react, so this works for test coverage and the reducer unit tests are good too. 

My comments are mostly cosmetic. Though if I came back to this code later, I'd probably look into the test if I didn't see an `expect` at the the end...since we're OK with this test, I think the cosmetic changes are worthwhile.